submodules/islandora_webform_ingest/includes/utilities.inc

```diff
@@ -496,40 +503,35 @@ function iwi_set_form_element_byhash(&$form, &$form_state, &$cloned_elements, $e
             case "tabpanel":
               $tab = $element->findElement($child_element_hash);
               if ($tab) {
-                if (!array_key_exists($child_element_hash, $cloned_elements)) {
-                  // If not already cloned clone it.
-                  $new_tab = clone $tab;
-
-                  $set_element_properties = function ($element) {
-                    $element->default_value = NULL;
-                  };
-                  // Set element properties on the descendants.
-                  $new_tab->eachDecendant($set_element_properties);
-
-                  $element->adopt($new_tab);
-                  $new_tab_drupal = $new_tab->toArray();
-
-                  drupal_array_set_nested_value($new_tab_drupal, array_merge($future_partial_path, array('#attributes')), $input_value['#attributes']);
-                  drupal_array_set_nested_value($new_tab_drupal, array_merge($future_partial_path, array('#default_value')), $input_value['#default_value']);
-
-                  $special_element_ref[] = $new_tab_drupal;
-
-                  // Appending the drupal form array back to where it belongs.
-                  // Lets store where do we put this one, so we can later just
-                  // fill out the values if we need to.
-                  $special_element_children = element_children($special_element_ref);
-                  $cloned_elements[$child_element_hash] = end($special_element_children);
+                $special_element_children = element_children($special_element_ref);
+                if (!isset($special_element_children[$offset])) {
+                  $lastkid = end($special_element_children);
+                  for ($i = $lastkid; $i < $offset; $i++) {
```

Comment:
I tested with the basic image MODS form, and the path includes a "0" (see screenshot), so yes, the MODS form does account for the difference. I'll see if I can modify the form parsing function to add the tab index.
![screen shot 2015-10-08 at 11 45 01 am](https://cloud.githubusercontent.com/assets/241240/10372015/09759682-6db4-11e5-9919-de765f557fa8.png)